plugin/evm/vm.go

```diff
@@ -445,23 +451,29 @@ func (vm *VM) Initialize(
 	default:
 		stepSize = 1 // as incremental as possible for testing purposes
 	}
-
-	// Initialize the validator transitioner, which is responsible for smoothly
-	// transitioning validators from the default set to the FTSO set, wrap it in
-	// a normalizer to have uniform weights across epochs, and wrap it in a cache
-	// to avoid unnecessary recomputation.
-	activeValidators := NewValidatorsTransitioner(ctx.Log, defaultValidators, cachedFTSOValidators,
+	transitionValidators := NewValidatorsTransitioner(ctx.Log, cacheActiveValidators, defaultValidators, cacheFTSOValidators, activeValidators,
 		WithStepSize(stepSize),
 	)
-	normalizedActiveValidators := NewValidatorsNormalizer(ctx.Log, activeValidators)
-	cachedNormalizedActiveValidators := NewValidatorsCache(normalizedActiveValidators,
-		WithCacheSize(uint(len(validators))),
-	)
+	normalizeTransitionValidators := NewValidatorsNormalizer(ctx.Log, transitionValidators)
+
+	// Getting the active validators for the current epoch will bootstrap all of the
+	// storage that is part of the transition logic, and will avoid long delays once
+	// we start processing blocks.
+	epoch, err := ftso.Current(lastAccepted.Hash())
+	if err != nil && !errors.Is(err, errFTSONotDeployed) && !errors.Is(err, errFTSONotActive) {
+		return fmt.Errorf("could not get current epoch: %w", err)
+	}
+	if err == nil {
+		_, err = normalizeTransitionValidators.ByEpoch(epoch)
```

Comment:
I didn't understand what exactly is happening here.

Reply:
It's described in the comment. It will execute the transitioner code, which will populate the validators store and all of the intermediate caches.